src/pages/preferences/preferencesTabs/publicKeysTab/EditModal.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from "react";
+import { Modal } from "components/Modal";
+import { useMutation } from "@apollo/react-hooks";
+import styled from "@emotion/styled";
+import Button, { Variant } from "@leafygreen-ui/button";
+import {
+  GetMyPublicKeysQuery,
+  UpdatePublicKeyMutation,
+  UpdatePublicKeyMutationVariables,
+  GetMyPublicKeysQueryVariables,
+  CreatePublicKeyMutation,
+  CreatePublicKeyMutationVariables,
+} from "gql/generated/types";
+import { GET_MY_PUBLIC_KEYS } from "gql/queries";
+import { UPDATE_PUBLIC_KEY } from "gql/mutations/update-public-key";
+import { useBannerDispatchContext } from "context/banners";
+import { Input } from "antd";
+import { CREATE_PUBLIC_KEY } from "gql/mutations/create-public-key";
+import { InputLabel, ErrorMessage } from "components/styles";
+import { v4 as uuid } from "uuid";
+
+const { TextArea } = Input;
+
+export interface EditModalPropsState {
+  replaceKeyName?: string;
+  initialKeyName?: string;
+  initialKeyValue?: string;
+  visible: boolean;
+  existingKeys?: {
+    name: string;
+    key?: string;
+  }[];
+}
+
+interface EditModalProps extends EditModalPropsState {
+  onCancel: () => void;
+}
+
+export const EditModal: React.FC<EditModalProps> = ({
+  replaceKeyName,
+  initialKeyName,
+  initialKeyValue,
+  existingKeys,
+  visible,
+  onCancel,
+}) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [errors, setErrors] = useState<string[]>([]);
+  const [updatePublicKey] = useMutation<
+    UpdatePublicKeyMutation,
+    UpdatePublicKeyMutationVariables
+  >(UPDATE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error editing the public key: ${error.message}`
+      );
+    },
+    update(cache, { data }) {
+      cache.writeQuery<GetMyPublicKeysQuery, GetMyPublicKeysQueryVariables>({
+        query: GET_MY_PUBLIC_KEYS,
+        data: { myPublicKeys: [...data.updatePublicKey] },
+      });
+    },
+  });
+  const [createPublicKey] = useMutation<
+    CreatePublicKeyMutation,
+    CreatePublicKeyMutationVariables
+  >(CREATE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error creating the public key: ${error.message}`
+      );
+    },
+    update(cache, { data }) {
+      cache.writeQuery<GetMyPublicKeysQuery, GetMyPublicKeysQueryVariables>({
+        query: GET_MY_PUBLIC_KEYS,
+        data: { myPublicKeys: [...data.createPublicKey] },
+      });
+    },
+  });
+
+  const [keyName, setKeyName] = useState(initialKeyName ?? "");
+  const [keyValue, setKeyValue] = useState(initialKeyValue ?? "");
+
+  useEffect(() => {
+    setKeyName(initialKeyName);
+  }, [initialKeyName]);
+
+  useEffect(() => {
+    setKeyValue(initialKeyValue);
+  }, [initialKeyValue]);
+
+  useEffect(() => {
+    const inputErrors = [];
+    if (!keyName) {
```

Comment:
If you cancel the modal (I had valid data in, but I don't think it matters) 2 of these 3 errors flash briefly in the error container. It's probably because there is a brief rerender in response to setting the props to their defaults before the modal is hidden. Can you add a check so that it doesn't happen?